### ISSUE TITLE
When annotations are displayed, the line height is increased

### ISSUE THREAD

**Issue by swongu** [jisaacks/GitGutter#523]
## Description

I have the following line configurations:
```
	"line_padding_bottom": -2,
	"line_padding_top": -2,
```
When annotations are displayed, the line height is increased. See below. I am using v1.9.4 with ST3 v3.1.1.

![screen shot 2018-10-04 at 11 33 43 am](https://user-images.githubusercontent.com/8647726/46495098-63b59a80-c7c9-11e8-9a2e-486df5a72a0b.png)

I would like to add that this annotation feature isn't mentioned on the documentation pages

## Support Info

- Sublime Text 3176
- Platform: osx
- Arch: x64
- GitGutter 1.9.4
- Install via PC: True
- git version 2.13.3
- mdpopups 3.3.3
- markdown 2.6.11
- pygments 2.1a0
- jinja2 2.8

## Steps to Reproduce Issue

1. Use the following line configurations:
```
	"line_padding_bottom": -2,
	"line_padding_top": -2,
```
2. Select a line that produces a Git annotation

**Comment by deathaxe** [comment]
This should happen with any phantom being added to such a line via any plugin as phantoms minimum height requires a positive padding value. All padding of phantoms was removed with 1.9.4 to avoid this issue with  `line_padding_bottom/top: 0`. But I guess there is no way to fix it for negative values.

Sorry guy but this is how phantoms just work. Nothing a plugin can do against it.

**Comment by fauxpark** [comment]
Sorry to bump, but I'm still getting this on Windows 10 with the default `line_padding` and Consolas 12pt.

If I leave the font size as the default 11, increase it to 14, or set `line_padding_bottom` to 1 the shifting doesn't happen.

For reference:

- Sublime Text 3176
- Platform: windows
- Arch: x64
- GitGutter 1.9.5
- Install via PC: True
- git version 2.20.1
- mdpopups 3.3.3
- markdown 2.6.11
- pygments 2.1a0
- jinja2 2.8

**Comment by deathaxe** [comment]
Sorry, but the bump is useless. This issue can't be fixed by GitGutter as stated with the "not possible with Sublime API" label.

**Comment by deathaxe** [comment]
A plugin can't influence how phantoms change the content of a view. Phantoms are canvas objects of html content, which are injected into the normal text content at a predefined text position. I made sure to remove any margin or padding from that html objects to avoid any kind of extra spaces around the text, which would cause the phantom to exceed the normal line height. This is all a plugin can do.

While working at the line annotation feature a couple of bugs and limitations with the phantom system were identified, which all need to be fixed by the coredevs @wbond or @jps. 

I tried my best to file detailed issue descriptions about everything I found.

We need to hope and wait.

**Comment by swongu** [comment]
> We need to hope and wait.
Any luck with the 3.2 release?

**Comment by sparr** [comment]
@deathaxe can you link to the issues in question?

**Comment by deathaxe** [comment]
Hmm, looks like I missed to report it. 

But I found a possibly too simple to see solution for that issue. Just reduced the `font-size` to `0.85rem`. I can now set `line_padding_top/bottom: 0` withouth line height changes on Windows.

Can anyone checkout the current master to confirm c4320af2ef0a78305514f78ce4eabf03b8599a7b fixes the issue for Linux/MacOS as well, or if there are remaining issues with probably some kind of fonts?

**Comment by deathaxe** [comment]
Should be fixed by 1.10.2
